### DISCUSSION TITLE
V7: variable fonts and missing characters/ligatures

### DISCUSSION THREAD

**Discussion by subframe7536** [subframe7536/maple-font#293] [General]
- [ ] landing page
- [x] make it variable with weight #27
- [x] some changes on characters
- [x] complete code range according to `Jetbrains Mono` #139
- [x] make it as stylesets and provide freezed version #108

[more planned items](https://github.com/users/subframe7536/projects/1)

**Comment by arjpar** [comment]
@subframe7536 When do you think V7 will be released?

**Comment by subframe7536** [comment]
There is no definite time. Probably in the middle of this year.

**Comment by viocha** [comment]
能否考虑支持一下多个等号和多个连字符的连体，下面是cacadia code的样式
![image](https://github.com/subframe7536/maple-font/assets/66580331/7b6f806e-0142-4734-89dd-80fb4241e106)

下面是maple font的现在的样式
![image](https://github.com/subframe7536/maple-font/assets/66580331/8c4cb36a-dd51-43bd-926e-3f55410fa3f8)

**Comment by subframe7536** [comment]
之前也考虑过，但是除了 markdown 表格外，我似乎想不到有什么应用场景

**Comment by viocha** [comment]
> 之前也考虑过，但是除了 markdown 表格外，我似乎想不到有什么应用场景

在注释里面用来分隔不同部分的代码，如果能连在一起，看起来更舒适

**Comment by subframe7536** [comment]
这似乎并不是一个常见的需求。改动起来有些麻烦。我需要考虑一下

**Comment by subframe7536** [comment]
in [monaspace v1.1](https://github.com/githubnext/monaspace/releases/tag/v1.100), they make the decision to break the long sequence equal or hyphen like `----` `====`

**Comment by subframe7536** [comment]
@0x0-00 粗斜体的字体不是maple mono

**Comment by subframe7536** [comment]
#198

**Comment by amsesk** [comment]
Missing icon: 💬

**Comment by mrChest228** [comment]
Please, add oo (infinity) ligature

**Comment by georgealbert** [comment]
请问如何调整行高，字符很漂亮，就是行间距大了，显示的行数有点少。

**Reply by subframe7536** [reply]
一般来说编辑器都可以调整行高

**Reply by georgealbert** [reply]
其实我是想调整glyph的Width和winAscent(win上高)和winDescent(win下深)、字形上高和字形下深、HHead上高和HHead下深。这样行距可以小点。

现在每个字符宽度是600，其实是有点宽了，请问怎么改？我把`MapleMono[wght].glyphs`里面的width = 600;改成width = 500; 没什么用啊。

**Reply by georgealbert** [reply]
和中文字体合并时，一般中文字体是1000的宽度，而maple西文字体是600的宽度，中英文就对不齐了。但是maple要是直接用fontforge改西文字体宽度的话，西文字符会挤在一起，不好看了。能像`Iosevka`一样可以自定义字体宽度就好了。

**Reply by subframe7536** [reply]
行高直接建议用 `ftcli` 改
https://ftcli.github.io/FoundryTools-CLI/commands/ftcli_os2.html#cmdoption-ftcli-os2-set-attrs-tasc
https://ftcli.github.io/FoundryTools-CLI/commands/ftcli_hhea.html#cmdoption-ftcli-hhea-ascent

关于字符宽度：https://github.com/subframe7536/maple-font/issues/211#issuecomment-2216314701

**Reply by subframe7536** [reply]
之前做过一版550宽度的，用了几天不太合适
https://github.com/subframe7536/maple-font/issues/131#issuecomment-2398540355

**Reply by subframe7536** [reply]
宽度轴 V7 是不打算做了，本身是基于 `JetBrains Mono`改的，没作业抄的话工作量有点大

**Reply by subframe7536** [reply]
至于手动改宽度，我也不太会。。。见 #249

**Reply by georgealbert** [reply]
> 至于手动改宽度，我也不太会。。。见 #249

我也是参考你的这段代码改的中英文宽度。但是按600的宽度，调整中文到1200，中文字符的间距就很宽，把maple的西文字体压缩到500，就挤在一起。

我现在是用`FantasqueSans`和`LXGW Lite Mono`或者`HarmonyOS SC`来合并字体。`FantasqueSans`调整upm到1000后，字符宽度是520，而中文从1000调整到1040，正好是西文字符的2倍，同时中文字符间距没有拉得很大，看起来比较舒服。

`FantasqueSans`是用`fontforge`生成的，sfdir格式，fontforge打开后调整upm，再用你的代码微调西文字符宽度，用魔兽字体合并工具合并后，再调整中文字符宽度。

也尝试过修改`FiraCode`的宽度，因为FiraCode本身太宽了，改到550左右显得有点挤。

`Iosevka`定制性非常好，也能做到中英文等宽对齐，但是本身是瘦高的字形，挤成一坨，看着非常累，用几分钟就要换个字体。

**Reply by subframe7536** [reply]
按照我的实验，`JetBrains Mono` 的字形宽度到 550 是极限了，因为不仅需要减小宽度，还需要缩放 X 轴，同时笔画宽度不能减小，很难自动处理

只能说目前是妥协的产物，编码这种多英文少中文的场景使用下来没啥大问题

**Reply by georgealbert** [reply]
1100的宽度对于中文来说，一下就宽了10%，就有点宽了。我看看`JetBrains Mono`。

**Reply by georgealbert** [reply]
<img width="649" alt="image" src="https://github.com/user-attachments/assets/bcc897b5-0a59-4272-ad77-253d6ce8d242">
我喜欢这个风格的字体。
